cmd/zed/internal/lakemanage/scan.go

```diff
@@ -66,20 +66,20 @@ from %q@%q:objects
 `
 
 type objectIterator struct {
-	reader      zio.ReadCloser
+	reader      zbuf.ProgressReadCloser
 	unmarshaler *zson.UnmarshalZNGContext
 	arena       *zed.Arena
 }
 
 func newObjectIterator(ctx context.Context, lake api.Interface, head *lakeparse.Commitish) (*objectIterator, error) {
 	query := fmt.Sprintf(iteratorQuery, head.Pool, head.Branch, head.Pool, head.Branch)
-	r, err := lake.Query(ctx, nil, query)
+	q, err := lake.Query(ctx, nil, query)
 	if err != nil {
 		return nil, err
 	}
 	arena := zed.NewArena()
 	return &objectIterator{
-		reader:      r,
+		reader:      zbuf.ScannerReader(q),
```

Comment:
Instead of adding zbuf.ScannerReader, how about

```suggestion
		puller:      q,
		reader:      zbuf.PullerReader(q),
```

plus

```go
func (r *objectIterator) close() {
	return r.puller.Pull(true)
}
```

plus

```go
	defer q.Pull(true)
	reader := zbuf.PullerReader(q)
```

in docs/library/go.md?

Reply:
sure